projects/rocr-runtime/libhsakmt/tests/kfdtest/src/KFDBaseComponentTest.cpp

```
@@ -427,13 +427,58 @@ HSAKMT_STATUS KFDBaseComponentTest::KFDTestMultiGPU(Test_Function test_function,
    return r;
 }
 
+HSAKMT_STATUS KFDBaseComponentTest::KFDTestMultiGPUStd(
+                                            std::function<void(int)> test_func, 
+                                            const std::vector<int>& gpuNodes,
+                                            unsigned int gpu_num) {
+    HSAKMT_STATUS r = HSAKMT_STATUS_SUCCESS;
+    std::vector<std::thread> threads;
+    threads.reserve(gpu_num);
+
+    if (gpuNodes.empty()) return HSAKMT_STATUS_SUCCESS;
+
+    for (int i = 0; i < gpu_num; i++) {
+        int gpu_node = gpuNodes.at(i);
+
+        try {
+            threads.emplace_back([test_func, gpu_node]() {
+                try {
+                    test_func(gpu_node);
+                } catch (...) {
+                    LOG() << "test failed at gpu" << gpu_node << std::endl;
```

The error it not propagated here. If one or more test fails here, the test will return success. AI provided an option to improve this using std::async and std::future. Some thing like below.

// ...existing code...
std::vector<std::future<void>> futures;

try {
    futures.push_back(std::async(std::launch::async, [test_func, gpu_node]() {
        test_func(gpu_node);  // Let exceptions propagate naturally
    }));
} catch (const std::system_error& e) {
    std::cout << "Thread creation for gpu node failed : " << gpu_node
              << " " << e.what() << std::endl;
    r = HSAKMT_STATUS_ERROR;
    break;
}

// ...existing code...

// Later, when joining threads, check for exceptions:
for (auto& future : futures) {
    try {
        future.get();  // Rethrows any exception from the thread
    } catch (const std::exception& e) {
        LOG() << "Test failed: " << e.what() << std::endl;
        r = HSAKMT_STATUS_ERROR;
    }
}

---

I think this is a google test framework related issue, it would not "catch" the errors. I will reform it in another way.